lenskit-core/src/main/java/org/grouplens/lenskit/data/history/History.java

```diff
@@ -75,6 +75,8 @@ private History() {}
      * @return An empty history for the user.
      */
     @Nonnull
+    @SafeVarargs
+    @SuppressWarnings("varargs") // method is safe MDE 2015-05-08
     public static <E extends Event> UserHistory<E> forUser(long id, E... events) {
         List<E> list = ImmutableList.copyOf(events);
```

Comment:
The exact set of warnings I am trying to silence here have to do with a problem called *heap corruption*. Exactly how a heap corruption arises is a complex story involving array covariance and type erasure. However, it is my understanding that it can only happen (as a result of our method) if our method somehow results in the array's contents being changed.

Since the three methods to which I applied this suppression only read the array, never mutate it, I believe that they are safe and the warnings can be safely suppressed. If it would help, I could expand my comments to state that it is safe because the arrays are never modified.